vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategy.java

```diff
@@ -65,7 +65,7 @@ public SecurityContext getContext() {
     @NonNull
     private Optional<SecurityContext> getFromVaadinSession() {
         VaadinSession session = VaadinSession.getCurrent();
-        if (session == null) {
+        if (session == null || session.getSession() == null) {
```

Comment:
tests ?

This can be easily revert back by someone accidentally without a test and no-one notice this ever

Reply:
Added